atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/priority/SweepPriorityStoreImpl.java

```diff
@@ -0,0 +1,144 @@
+import com.palantir.processors.AutoDelegate;
+
+@AutoDelegate(typeToExtend = SweepPriorityStore.class)
+public final class SweepPriorityStoreImpl implements SweepPriorityStore {
+    private class InitializingWrapper extends AsyncInitializer implements AutoDelegate_SweepPriorityStore {
+
+        @Override
+        public SweepPriorityStoreImpl delegate() {
+            checkInitialized();
+            return SweepPriorityStoreImpl.this;
+        }
+        @Override
+        protected void tryInitialize() {
+            SweepPriorityStoreImpl.this.tryInitialize();
+        }
+
+        @Override
+        protected String getInitializingClassName() {
+            return "SweepPriorityStore";
+        }
+
+    }
+
+    private final KeyValueService kvs;
+    private final  SweepTableFactory sweepTableFactory;
+    private InitializingWrapper wrapper = new InitializingWrapper();
+
+    private SweepPriorityStoreImpl(KeyValueService kvs, SweepTableFactory sweepTableFactory) {
+        this.kvs = kvs;
+        this.sweepTableFactory = sweepTableFactory;
+    }
+
+    public static SweepPriorityStore create(KeyValueService kvs, SweepTableFactory sweepTableFactory,
+            boolean initializeAsync) {
+        SweepPriorityStoreImpl sweepPriorityStore = new SweepPriorityStoreImpl(kvs, sweepTableFactory);
+        sweepPriorityStore.wrapper.initialize(initializeAsync);
+        return sweepPriorityStore.wrapper.isInitialized() ? sweepPriorityStore : sweepPriorityStore.wrapper;
+    }
+
+    @Override
+    public List<SweepPriority> loadOldPriorities(Transaction tx, long sweepTimestamp) {
+        return loadPriorities(new SweepPriorityTransaction(tx, sweepTimestamp));
+    }
+
+    @Override
+    public List<SweepPriority> loadNewPriorities(Transaction tx) {
+        return loadPriorities(tx);
+    }
+
+    @Override
+    public void update(Transaction tx, TableReference tableRef, UpdateSweepPriority update) {
+        SweepPriorityRow row = SweepPriorityRow.of(tableRef.getQualifiedName());
+        SweepPriorityTable table = sweepTableFactory.getSweepPriorityTable(tx);
+        update.newStaleValuesDeleted().ifPresent(n -> table.putCellsDeleted(row, n));
+        update.newCellTsPairsExamined().ifPresent(n -> table.putCellsExamined(row, n));
+        update.newLastSweepTimeMillis().ifPresent(t -> table.putLastSweepTime(row, t));
+        update.newMinimumSweptTimestamp().ifPresent(t -> table.putMinimumSweptTimestamp(row, t));
+        update.newWriteCount().ifPresent(c -> table.putWriteCount(row, c));
+    }
+
+    @Override
+    public void delete(Transaction tx, Collection<TableReference> tableRefs) {
+        sweepTableFactory.getSweepPriorityTable(tx).delete(
+                Collections2.transform(tableRefs, tr -> SweepPriorityRow.of(tr.getQualifiedName())));
+    }
+
+    private void tryInitialize() {
+        Schemas.createTablesAndIndexes(SweepSchema.INSTANCE.getLatestSchema(), kvs);
```

Comment:
Hmm, this is technically correct given that the sweep schema now only contains one table, but might be awkward if the sweep schema contained other tables. A little conflicted on the naming but I think this is fine, wouldn't block on figuring this out.